pkg/service/init.go

```diff
@@ -127,7 +133,41 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, _
 
 	// Very important that this bootstrap handler is called after the NewServiceMetrics handler so
 	// MetricsManager dependency has been created.
-	common.InitializeSentMetrics(s.lc, dic)
+	sdkCommon.InitializeSentMetrics(s.lc, dic)
+	return true
+}
+
+func (b *Bootstrap) checkDependencyServiceAvailable(serviceKey string, startupTimer startup.Timer) bool {
+	lc := b.deviceService.lc
+	registry := bootstrapContainer.RegistryFrom(b.deviceService.dic.Get)
+	mode := bootstrapContainer.DevRemoteModeFrom(b.deviceService.dic.Get)
+	clients := bootstrapContainer.ConfigurationFrom(b.deviceService.dic.Get).GetBootstrap().Clients
+	pingUrl := (*clients)[serviceKey].Url() + common.ApiPingRoute
```

Comment:
```suggestion
	clientInfo, ok := (*clients)[serviceKey]
	if !ok {
	  lc.Errorf("Client configuration for '%s' not found, missing common config? Use -cp or -cc flags for common config.", serviceKey)
	}
	pingUrl := clientInfo.Url() + common.ApiPingRoute
```

this function might be copied to other places, so add the check to prevent panic